@navikt/core/css/modal.css

```diff
@@ -1,69 +1,114 @@
+  border-radius: var(--a-border-radius-large);
   box-shadow: var(--a-shadow-xlarge);
+  padding: 0;
+  position: fixed;
 }
 
-.navds-modal__content {
-  padding: var(--a-spacing-4);
+.navds-modal[open] {
+  display: flex;
+  flex-direction: column;
 }
 
-.navds-modal:focus-visible,
-.navds-modal--focus {
-  box-shadow: var(--a-shadow-focus);
-  outline: none;
+.navds-modal--polyfilled {
+  top: 50%;
+  transform: translate(0, -50%);
+  overflow: auto;
+  max-height: 80%;
+
+  /* From polyfill (dialog-polyfill/dist/dialog-polyfill.css): */
+  left: 0;
+  right: 0;
+  width: fit-content;
+  height: fit-content;
+  margin: auto;
 }
 
-@supports not selector(:focus-visible) {
-  .navds-modal:focus,
-  .navds-modal--focus {
-    box-shadow: var(--a-shadow-focus);
-    outline: none;
-  }
+.navds-modal--polyfilled:not([open]) {
+  display: none; /* from polyfill */
 }
 
-.navds-modal__overlay {
+._dialog_overlay {
+  /* From polyfill */
   position: fixed;
-  z-index: var(--a-z-index-modal);
   top: 0;
+  right: 0;
   bottom: 0;
   left: 0;
-  right: 0;
-  background-color: var(--ac-modal-backdrop, var(--a-surface-backdrop));
-  padding: var(--a-spacing-4);
-  display: flex;
-  align-items: center;
-  justify-content: center;
 }
 
-.navds-modal__button--shake {
-  transform: rotate(0deg);
-  transition: transform 100ms cubic-bezier(0.82, 2, 1, 0.3);
+@media (min-width: 768px) {
+  .navds-modal {
+    max-width: 700px;
+  }
+}
+
+@media (max-width: 767px) {
+  .navds-modal {
+    max-height: 100%;
+    max-width: 100%;
+  }
+}
+
+.navds-modal::backdrop {
+  /*
+   * Cannot use --a-surface-backdrop because ::backdrop does not inherit from anything but itself.
+   * But consumers can do `::backdrop { --ac-modal-backdrop: white; }`
+   * We can add ::backdrop to tokens.css when we remove the polyfill.
+   */
+  background: var(--ac-modal-backdrop, rgb(0 0 0 / 0.44));
 }
 
-/* prettier-ignore */
-.navds-modal__overlay:active > .navds-modal:not(:active) > .navds-modal__button--shake {
-  transform: rotate(10deg);
-  transition: transform 100ms cubic-bezier(0.82, -2, 1, 0.3);
+.navds-modal--polyfilled + .backdrop /* Cannot be combined with ::backdrop selector */ {
+  background: var(--ac-modal-backdrop, rgb(0 0 0 / 0.44));
+
+  /* From polyfill: */
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
 }
 
 .navds-modal__button {
-  position: absolute;
-  top: var(--a-spacing-4);
-  right: var(--a-spacing-4);
+  margin-left: var(--a-spacing-4);
+  float: right;
+}
+
+.navds-modal__header {
+  padding: var(--a-spacing-6) var(--a-spacing-6) var(--a-spacing-4) var(--a-spacing-6);
+}
+
+.navds-modal__header-icon svg {
+  display: inline;
+  vertical-align: -0.25rem;
+  margin-right: var(--a-spacing-1);
+}
+
+.navds-modal__label {
+  font-weight: 600;
+  color: var(--a-text-subtle);
+}
+
+.navds-modal__content {
+  padding: var(--a-spacing-6);
+  overflow: auto;
+  overscroll-behavior: contain;
```

Comment:
Er dette en mulighet vi bør tilby vår modal? 
https://atlassian.design/components/modal-dialog/examples#scrolling-behavior

Reply:
Er ikke så kjent med hvilke behov som finnes, men hvis det er behov for det så bør vi sikkert det. Ønsker imidlertid ikke å legge til kompleksitet unødvendig. Tenker du mest på det å kunne ha header og footer som ikke er sticky, eller også på det at scrollbaren er på backdropen slik at hele modalen scrolles i stedet for bare innholdet? Har en følelse av at sistnevnte kan bli komplisert, men har ikke testet. Hvis vi bare trenger en variant det header og footer ikke er sticky tror jeg det er enkelt å bare flytte scroll til `<dialog>`-elementet.